packages/markers/src/common/problem-marker.ts

```diff
@@ -24,7 +25,7 @@ export interface ProblemMarker extends Marker<Diagnostic> {
 }
 
 export namespace ProblemMarker {
-    export function is(node: Marker<object>): node is ProblemMarker {
-        return 'kind' in node && node.kind === PROBLEM_KIND;
+    export function is(node: unknown): node is ProblemMarker {
+        return isObject<Marker<object>>(node) && node.kind === PROBLEM_KIND;
```

Comment:
I'm not sure this is correct: shouldn't we assert it's a marker first? Right now an object like this wold fit the bill: 
```
{
   kind: 'problem'
}
```

would fit the bill, even though it's not a marker. But asserting `ProblemMarker.is(foo)` implies that it's a marker, typing-wise.

Reply:
@tsmaeder I agree, the code is very optimistic. But the previous code did the same assumption.
So I would suggest to tackle this improved marker assertion in a separate ticket.

Reply:
Well, during the runtime, yes. But from a statical perspective, it assumed that the `node` already is a `Marker<object>`. So we should confirm that as well.

Reply:
I think the static typing problem stems from https://github.com/eclipse-theia/theia/issues/12253. That is a considerably bigger fish to fry, though, and it would be asking to much from a PR to fix a concrete problem. @msujew what are you suggesting, concretely?